Portfolio: Mobile menu

osmanwako opened the pull request:
#  Implementation of Hamburger Menu 🍔 
- The following changes have been made to implement a hamburger menu: 
-  Display the navigation links by clicking the hamburger menu icon.  
- Opened hamburger menu closes on click of the ❌  - icon.  
- Clicking navigation links results in:
  - Navigation to the required area. ✔️ 
  - Closes the hamburger menu. ✔️
-  On bigger screens the hamburger icon is hidden and the navigation links appear normal. ✔️

osmanwako committed:
update attribute to html elements on  index html file

osmanwako committed:
update attribute to html elements on  index html file

Bennyjoez committed:
Add Javascript functionality to popup menu

Bennyjoez committed:
Add Javascript functionality to popup menu

osmanwako committed:
Add linter file

osmanwako committed:
Merge branch 'master' into feature_mobilepopupmenu

osmanwako committed:
fixed linter error

osmanwako committed:
fixed linter error

osmanwako committed:
javascript listener added

osmanwako committed:
Update linters.yml

osmanwako committed:
Fix linter errors

osmanwako committed:
Update javascript.js

osmanwako committed:
fix error

osmanwako committed:
fixed error

osmanwako committed:
Fix linters error

osmanwako committed:
Fix lightouse

osmanwako committed:
Fix lightouse + STYLELINT

osmanwako committed:
linter error

osmanwako committed:
linter errors

brainconnect93 commented on assets/style/main.css:
```diff
@@ -476,14 +484,56 @@ header {
 }
 
 @media (max-width: 768px) {
+  .popupmenu-container {
+    width: 100vw;
+    height: 100vh;
+    position: fixed;
+    top: 0;
+    left: 0;
+    background-color: rgba(96, 112, 255, 0.95);
+    mix-blend-mode: multiply;
+    padding-left: 10px;
+    padding-top: 60px;
+    border: white 1px solid;
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    align-items: stretch;
+  }
+
```
- [ ] Hi @osmanwako, Great  effort working on the mobile menu of your portfolio project, But i noticed somethings that needs attention to work on. when the hamburger is click and the menu displayed i notice some extra space with different color my the bottom and right hand side. I would kindly suggest you change the width and height of the container from `vw`,`vh` to `%` in order to remove the extra spaces.
- [ ] Also, in the Figma design when the mobile menu is displayed, the main page at the background becomes `blur` but in your own case the main page is not blur, kindly check through your styling and try to implement the blurring of the main page when the mobile menu is displayed.


| Your Design 👇   | Figma Design 👇  |
| ------------- | ------------- |
| ![Osman-mob](https://user-images.githubusercontent.com/78634841/223665527-d8ce79af-bbb0-446d-89cf-6fe55bc90d24.jpg)  | ![mob-orig](https://user-images.githubusercontent.com/78634841/223665610-e40e7797-b8f4-40f4-bf85-52ccbb2f6400.png)  |

brainconnect93 reviewed (changes requested):
# STATUS: Changes Required :recycle:

### Hi @osmanwako, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media.tenor.com/isiLRnxz3zwAAAAM/cat-driving-serious.gif" width="500px" height="300px">

### To Highlight :1st_place_medal: 
-  Has a proper Pull Request. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Github flow ✔️ 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Bennyjoez committed:
Respond to review changes

> Fixed extra space in the popup menu as requested.
> Blur background added to the popup menu.

roshan-bajgain reviewed (approved):
Hi @osmanwako, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Requested change ✔️ 
- Mobile menu is working fine ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
